presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PullConstantsAboveGroupBy.java

```diff
@@ -62,7 +63,7 @@ public class PullConstantsAboveGroupBy
     @Override
     public boolean isEnabled(Session session)
     {
-        return isOptimizeConstantGroupingKeys(session);
+        return isOptimizeConstantGroupingKeys(session) || isRewriteExpressionWithConstantEnabled(session);
```

Comment:
Why di we need this change?

Reply:
Because the new optimizer added is to extract constant only, it relies on this PullConstantsAboveGroupBy to do the optimization. Hence we need to make sure that when this optimizer is enabled, PullConstantsAboveGroupBy is also enabled.